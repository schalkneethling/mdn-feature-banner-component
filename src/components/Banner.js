// @flow

import React from "react";

const Banner = (props: {
  customClasses: string,
  targetURL: string,
  calloutMainCopy: string,
  calloutSubtext: string,
  buttonCopy: string
}) => {
  return (
    <div className={`column-callout ${props.customClasses}`}>
      <a href={props.targetURL}>
        <span>
          <p className="callout-main-copy">
            {props.calloutMainCopy}
            <span className="callout-subtext">{props.calloutSubtext}</span>
          </p>
          <button type="button">{props.buttonCopy}</button>
        </span>
      </a>
    </div>
  );
};

export default Banner;
