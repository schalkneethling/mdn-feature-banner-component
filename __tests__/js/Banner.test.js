import React from "react";
import { create } from "react-test-renderer";
import Banner from "../../src/components/Banner";

test("banner snapshot", () => {
  const banner = create(
    <Banner
      targetURL="https://donate.mozilla.org"
      customClasses="callout-foundation"
      calloutMainCopy="Mozilla"
      calloutSubtext="Proudly non-profit. Donate before Dec 31."
      buttonCopy="Donate Now"
    />
  );
  expect(banner.toJSON()).toMatchSnapshot();
});
