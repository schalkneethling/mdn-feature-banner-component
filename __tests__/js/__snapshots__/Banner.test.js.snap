// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`banner snapshot 1`] = `
<div
  className="column-callout callout-foundation"
>
  <a
    href="https://donate.mozilla.org"
  >
    <span>
      <p
        className="callout-main-copy"
      >
        Mozilla
        <span
          className="callout-subtext"
        >
          Proudly non-profit. Donate before Dec 31.
        </span>
      </p>
      <button
        type="button"
      >
        Donate Now
      </button>
    </span>
  </a>
</div>
`;
